packages/router/src/util.ts

```diff
@@ -103,24 +103,23 @@ const matchPath = (
   // Get the names and the transform types for the given route.
   const routeParams = paramsForRoute(route)
   const allParamTypes = { ...coreParamTypes, ...paramTypes }
-  let typeConstrainedRoute = route
+  let typeMatchingRoute = route
 
-  // Map all params from the route to their type constraint regex to create a
-  // "type-constrained route" regex
+  // Map all params from the route to their type `match` regexp to create a
```

Comment:
@mojombo I know I said I didn't care if these were "regex" or "regexp", but I just now realized JS has already decided for us. It's "regexp" (as in `new RegExp(`^${typeMatchingRoute}$`, 'g')` for example)

Reply:
Fair enough!